jujugui/static/gui/src/app/store/env/sandbox.js

```diff
@@ -1346,15 +1353,99 @@ YUI.add('juju-env-sandbox', function(Y) {
     },
 
     /**
+      Handle Cloud.UpdateCredentials messages.
+
+      @method handleCloudUpdateCredentials
+      @param {Object} data The contents of the API arguments.
+      @param {Object} client The active ClientConnection.
+      @param {Object} state An instance of FakeBackend.
+    */
+    handleCloudUpdateCredentials: function(data, client, state) {
+      const entries = data.params.credentials;
```

Comment:
Yes, but since the GUI itself sends the request (which is being parsed here) we can assume request params are properly built with 100% safety.